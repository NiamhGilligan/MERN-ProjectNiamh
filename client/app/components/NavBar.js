import React, { Component } from 'react';
import 'whatwg-fetch';

import { Link } from 'react-router-dom';


class NavBar extends Component {

  render() {


    var navStyle = {
      WebkitBoxShadow: "0 0 4px rgba(0,0,0,0.4)",
      MozBoxShadow: "0 0 4px rgba(0,0,0,0.4)",
      boxShadow : "0 0 4px rgba(0,0,0,0.4)",

    };



    var titleStyle = {};
    var linkStyle = {};

    if (this.props.bgColor)
      navStyle.background = this.props.bgColor;

    if (this.props.titleColor)
      titleStyle.color = this.props.titleColor;

    if (this.props.linkColor)
      linkStyle.color = this.props.linkColor;
    return (
      <div  >
        <nav style={navStyle} className = "navbar navbar-default">
          <div className="navbar-header">
           <Link style={titleStyle} className="navbar-brand" to="/">Teachers Toolbox</Link>
          </div>

          <div className = "collapse navbar-collapse" id="nav-collapse">
            <ul  className="nav navbar-nav" >

              <li><Link to="/contact" > Contact </Link></li>
              <li><Link to="/create"> Create </Link></li>
              <li> <Link to="/Login"> Login </Link></li>
              <li><Link to="/Register"> Register </Link></li>
              <li><Link to="/MyAccount"> My Account </Link></li>
              <li><Link to="/search"> Search </Link></li>

            </ul>
          </div>
        </nav>
      </div>


    );
  }
}

export default NavBar;
