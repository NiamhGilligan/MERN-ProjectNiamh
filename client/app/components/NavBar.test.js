import React from 'react';
import {shallow} from 'enzyme';
import NavBar from './NavBar';

describe('Login Test', ()=> {

  test('Rendering Login page',() => {
    const wrapper = shallow(
      <NavBar/>
    );

    expect(wrapper).toMatchSnapshot();
  });


});
