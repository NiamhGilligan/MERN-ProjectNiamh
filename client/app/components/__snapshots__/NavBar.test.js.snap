// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Test Rendering Login page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="navbar navbar-default"
        style={
          Object {
            "MozBoxShadow": "0 0 4px rgba(0,0,0,0.4)",
            "WebkitBoxShadow": "0 0 4px rgba(0,0,0,0.4)",
            "boxShadow": "0 0 4px rgba(0,0,0,0.4)",
          }
        }
      >
        <div
          className="navbar-header"
        >
          <Link
            className="navbar-brand"
            replace={false}
            style={Object {}}
            to="/"
          >
            Teachers Toolbox
          </Link>
        </div>
        <div
          className="collapse navbar-collapse"
          id="nav-collapse"
        >
          <ul
            className="nav navbar-nav"
          >
            <li>
              <Link
                replace={false}
                to="/contact"
              >
                 Contact 
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/create"
              >
                 Create 
              </Link>
            </li>
            <li>
               
              <Link
                replace={false}
                to="/Login"
              >
                 Login 
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/Register"
              >
                 Register 
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/MyAccount"
              >
                 My Account 
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/search"
              >
                 Search 
              </Link>
            </li>
          </ul>
        </div>
      </nav>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="navbar-header"
          >
            <Link
              className="navbar-brand"
              replace={false}
              style={Object {}}
              to="/"
            >
              Teachers Toolbox
            </Link>
          </div>,
          <div
            className="collapse navbar-collapse"
            id="nav-collapse"
          >
            <ul
              className="nav navbar-nav"
            >
              <li>
                <Link
                  replace={false}
                  to="/contact"
                >
                   Contact 
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/create"
                >
                   Create 
                </Link>
              </li>
              <li>
                 
                <Link
                  replace={false}
                  to="/Login"
                >
                   Login 
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/Register"
                >
                   Register 
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/MyAccount"
                >
                   My Account 
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/search"
                >
                   Search 
                </Link>
              </li>
            </ul>
          </div>,
        ],
        "className": "navbar navbar-default",
        "style": Object {
          "MozBoxShadow": "0 0 4px rgba(0,0,0,0.4)",
          "WebkitBoxShadow": "0 0 4px rgba(0,0,0,0.4)",
          "boxShadow": "0 0 4px rgba(0,0,0,0.4)",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="navbar-brand"
              replace={false}
              style={Object {}}
              to="/"
            >
              Teachers Toolbox
            </Link>,
            "className": "navbar-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Teachers Toolbox",
              "className": "navbar-brand",
              "replace": false,
              "style": Object {},
              "to": "/",
            },
            "ref": null,
            "rendered": "Teachers Toolbox",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="nav navbar-nav"
            >
              <li>
                <Link
                  replace={false}
                  to="/contact"
                >
                   Contact 
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/create"
                >
                   Create 
                </Link>
              </li>
              <li>
                 
                <Link
                  replace={false}
                  to="/Login"
                >
                   Login 
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/Register"
                >
                   Register 
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/MyAccount"
                >
                   My Account 
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/search"
                >
                   Search 
                </Link>
              </li>
            </ul>,
            "className": "collapse navbar-collapse",
            "id": "nav-collapse",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <Link
                    replace={false}
                    to="/contact"
                  >
                     Contact 
                  </Link>
                </li>,
                <li>
                  <Link
                    replace={false}
                    to="/create"
                  >
                     Create 
                  </Link>
                </li>,
                <li>
                   
                  <Link
                    replace={false}
                    to="/Login"
                  >
                     Login 
                  </Link>
                </li>,
                <li>
                  <Link
                    replace={false}
                    to="/Register"
                  >
                     Register 
                  </Link>
                </li>,
                <li>
                  <Link
                    replace={false}
                    to="/MyAccount"
                  >
                     My Account 
                  </Link>
                </li>,
                <li>
                  <Link
                    replace={false}
                    to="/search"
                  >
                     Search 
                  </Link>
                </li>,
              ],
              "className": "nav navbar-nav",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/contact"
                  >
                     Contact 
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": " Contact ",
                    "replace": false,
                    "to": "/contact",
                  },
                  "ref": null,
                  "rendered": " Contact ",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/create"
                  >
                     Create 
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": " Create ",
                    "replace": false,
                    "to": "/create",
                  },
                  "ref": null,
                  "rendered": " Create ",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <Link
                      replace={false}
                      to="/Login"
                    >
                       Login 
                    </Link>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": " Login ",
                      "replace": false,
                      "to": "/Login",
                    },
                    "ref": null,
                    "rendered": " Login ",
                    "type": [Function],
                  },
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/Register"
                  >
                     Register 
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": " Register ",
                    "replace": false,
                    "to": "/Register",
                  },
                  "ref": null,
                  "rendered": " Register ",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/MyAccount"
                  >
                     My Account 
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": " My Account ",
                    "replace": false,
                    "to": "/MyAccount",
                  },
                  "ref": null,
                  "rendered": " My Account ",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/search"
                  >
                     Search 
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": " Search ",
                    "replace": false,
                    "to": "/search",
                  },
                  "ref": null,
                  "rendered": " Search ",
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "div",
        },
      ],
      "type": "nav",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="navbar navbar-default"
          style={
            Object {
              "MozBoxShadow": "0 0 4px rgba(0,0,0,0.4)",
              "WebkitBoxShadow": "0 0 4px rgba(0,0,0,0.4)",
              "boxShadow": "0 0 4px rgba(0,0,0,0.4)",
            }
          }
        >
          <div
            className="navbar-header"
          >
            <Link
              className="navbar-brand"
              replace={false}
              style={Object {}}
              to="/"
            >
              Teachers Toolbox
            </Link>
          </div>
          <div
            className="collapse navbar-collapse"
            id="nav-collapse"
          >
            <ul
              className="nav navbar-nav"
            >
              <li>
                <Link
                  replace={false}
                  to="/contact"
                >
                   Contact 
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/create"
                >
                   Create 
                </Link>
              </li>
              <li>
                 
                <Link
                  replace={false}
                  to="/Login"
                >
                   Login 
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/Register"
                >
                   Register 
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/MyAccount"
                >
                   My Account 
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/search"
                >
                   Search 
                </Link>
              </li>
            </ul>
          </div>
        </nav>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="navbar-header"
            >
              <Link
                className="navbar-brand"
                replace={false}
                style={Object {}}
                to="/"
              >
                Teachers Toolbox
              </Link>
            </div>,
            <div
              className="collapse navbar-collapse"
              id="nav-collapse"
            >
              <ul
                className="nav navbar-nav"
              >
                <li>
                  <Link
                    replace={false}
                    to="/contact"
                  >
                     Contact 
                  </Link>
                </li>
                <li>
                  <Link
                    replace={false}
                    to="/create"
                  >
                     Create 
                  </Link>
                </li>
                <li>
                   
                  <Link
                    replace={false}
                    to="/Login"
                  >
                     Login 
                  </Link>
                </li>
                <li>
                  <Link
                    replace={false}
                    to="/Register"
                  >
                     Register 
                  </Link>
                </li>
                <li>
                  <Link
                    replace={false}
                    to="/MyAccount"
                  >
                     My Account 
                  </Link>
                </li>
                <li>
                  <Link
                    replace={false}
                    to="/search"
                  >
                     Search 
                  </Link>
                </li>
              </ul>
            </div>,
          ],
          "className": "navbar navbar-default",
          "style": Object {
            "MozBoxShadow": "0 0 4px rgba(0,0,0,0.4)",
            "WebkitBoxShadow": "0 0 4px rgba(0,0,0,0.4)",
            "boxShadow": "0 0 4px rgba(0,0,0,0.4)",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="navbar-brand"
                replace={false}
                style={Object {}}
                to="/"
              >
                Teachers Toolbox
              </Link>,
              "className": "navbar-header",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Teachers Toolbox",
                "className": "navbar-brand",
                "replace": false,
                "style": Object {},
                "to": "/",
              },
              "ref": null,
              "rendered": "Teachers Toolbox",
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ul
                className="nav navbar-nav"
              >
                <li>
                  <Link
                    replace={false}
                    to="/contact"
                  >
                     Contact 
                  </Link>
                </li>
                <li>
                  <Link
                    replace={false}
                    to="/create"
                  >
                     Create 
                  </Link>
                </li>
                <li>
                   
                  <Link
                    replace={false}
                    to="/Login"
                  >
                     Login 
                  </Link>
                </li>
                <li>
                  <Link
                    replace={false}
                    to="/Register"
                  >
                     Register 
                  </Link>
                </li>
                <li>
                  <Link
                    replace={false}
                    to="/MyAccount"
                  >
                     My Account 
                  </Link>
                </li>
                <li>
                  <Link
                    replace={false}
                    to="/search"
                  >
                     Search 
                  </Link>
                </li>
              </ul>,
              "className": "collapse navbar-collapse",
              "id": "nav-collapse",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li>
                    <Link
                      replace={false}
                      to="/contact"
                    >
                       Contact 
                    </Link>
                  </li>,
                  <li>
                    <Link
                      replace={false}
                      to="/create"
                    >
                       Create 
                    </Link>
                  </li>,
                  <li>
                     
                    <Link
                      replace={false}
                      to="/Login"
                    >
                       Login 
                    </Link>
                  </li>,
                  <li>
                    <Link
                      replace={false}
                      to="/Register"
                    >
                       Register 
                    </Link>
                  </li>,
                  <li>
                    <Link
                      replace={false}
                      to="/MyAccount"
                    >
                       My Account 
                    </Link>
                  </li>,
                  <li>
                    <Link
                      replace={false}
                      to="/search"
                    >
                       Search 
                    </Link>
                  </li>,
                ],
                "className": "nav navbar-nav",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      replace={false}
                      to="/contact"
                    >
                       Contact 
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": " Contact ",
                      "replace": false,
                      "to": "/contact",
                    },
                    "ref": null,
                    "rendered": " Contact ",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      replace={false}
                      to="/create"
                    >
                       Create 
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": " Create ",
                      "replace": false,
                      "to": "/create",
                    },
                    "ref": null,
                    "rendered": " Create ",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      <Link
                        replace={false}
                        to="/Login"
                      >
                         Login 
                      </Link>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": " Login ",
                        "replace": false,
                        "to": "/Login",
                      },
                      "ref": null,
                      "rendered": " Login ",
                      "type": [Function],
                    },
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      replace={false}
                      to="/Register"
                    >
                       Register 
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": " Register ",
                      "replace": false,
                      "to": "/Register",
                    },
                    "ref": null,
                    "rendered": " Register ",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      replace={false}
                      to="/MyAccount"
                    >
                       My Account 
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": " My Account ",
                      "replace": false,
                      "to": "/MyAccount",
                    },
                    "ref": null,
                    "rendered": " My Account ",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      replace={false}
                      to="/search"
                    >
                       Search 
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": " Search ",
                      "replace": false,
                      "to": "/search",
                    },
                    "ref": null,
                    "rendered": " Search ",
                    "type": [Function],
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            "type": "div",
          },
        ],
        "type": "nav",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
